decode/data.ts

```diff
@@ -35,8 +35,11 @@ export function decode(tokens: Token[]): DataEvent {
                 targetData.push(target);
             }
             return { time, event, data: targetData };
+        case Event.Upgrade:
+            let upgrade: UpgradeData = { key: tokens[2] as string };
+            return { time, event, data: upgrade };
         case Event.Upload:
-            let upload: UploadData = { sequence: tokens[2] as number, attempts: tokens[3] as number, status: tokens[4] as number};
+            let upload: UploadData = { sequence: tokens[2] as number, attempts: tokens[3] as number, status: tokens[4] as number };
             return { time, event, data: upload };
         case Event.Metric:
```

Comment:
Add a default clause here in case you miss something. Similarly for all the big case of event clauses in the program (check them all).

Reply:
It's a good point. I'll cover data quality changes in another round. If I add default, I will need to figure out what to do with these unhandled cases. Probably throwing an exception is ok but worth thinking about more.

---
In reply to: [368662595](https://github.com/microsoft/clarity-js/pull/166#discussion_r368662595) [](ancestors = 368662595)